components/engine/src/mutable.rs

```diff
@@ -22,3 +22,23 @@ pub trait Mutable: Writable {
         Ok(())
     }
 }
+
+pub trait Mutable: WriteBatchBase {
```

Comment:
Why is the Self bound `WriteBatchBase`?

Reply:
There may be a different type which both implement `WriteBatchBase`. I will make a PR to `rust-rocksdb` for this later.

Reply:
At the beginning , I try such code:
```
impl WriteBatchBase for WriteBatch {
}
impl Mutable for WriteBatchBase {
}
```
But it seems that the trait `Mutable` was not implemented by  `WriteBatch` automatically.
So I change it into:
```
impl WriteBatchBase for WriteBatch {
}
pub trait Mutable: WriteBatchBase {
}
impl Mutable for WriteBatch {}
```